src/mongo/db/storage/csv_file.h

```diff
@@ -99,18 +101,23 @@ class CsvFileInput : public StreamableInput {
      * @param line read from csv, "data1,data2,data3..."
      * @return: vector containing each field as string, {"data1","data2","data3"...}.
      */
-    std::vector<std::string> parseLine(const std::string& line);
+    std::vector<std::string_view> parseLine(const std::string_view& line);
 
     template <CsvFieldType T>
-    void appendTo(BSONObjBuilder& builder, const std::string& fieldName, const std::string& data);
+    void appendTo(BSONObjBuilder& builder,
+                  const std::string& fieldName,
+                  const std::string_view& field);
 
     // Reads each line from the CSV file and converts it into a BSONObj, being compliant with the
     // metadata. It will return boost::none if there is no more line to read in the csv file.
     boost::optional<BSONObj> readBsonObj();
 
     std::string _fileAbsolutePath;
     std::string _metadataAbsolutePath;
-    std::ifstream _ifs;
+    int _fd = -1;
+    size_t _fileSize = 0;
+    const char* _data = nullptr;
+    size_t _offset = 0;
```

Comment:
What exactly is _fd for?
I figure that _fileSize is the size of the csv file,
_data is the mapped memory that holds content of the file in RAM, and _offset is the cursor that represents where we are reading in the file?

Reply:
`fd` is a common acronym for a file descriptor.